app/experimenter/nimbus-ui/src/components/FormMetrics/index.tsx

```diff
@@ -3,51 +3,102 @@
-  probeSets: (getConfig_nimbusConfig_probeSets | null)[] | null;
   isLoading: boolean;
   isServerValid: boolean;
   submitErrors: Record<string, string[]>;
+  setSubmitErrors: React.Dispatch<React.SetStateAction<Record<string, any>>>;
   onSave: (data: Record<string, any>, reset: Function) => void;
   onNext: (ev: React.FormEvent) => void;
 };
 
+type ProbeSet =
+  | getExperiment_experimentBySlug_primaryProbeSets
+  | getExperiment_experimentBySlug_secondaryProbeSets;
+
+type ProbeSets =
+  | (getExperiment_experimentBySlug_primaryProbeSets | null)[]
+  | (getExperiment_experimentBySlug_secondaryProbeSets | null)[];
+
 const FormMetrics = ({
   experiment,
-  probeSets,
   isLoading,
   isServerValid,
   submitErrors,
+  setSubmitErrors,
   onSave,
   onNext,
 }: FormMetricsProps) => {
-  const { handleSubmit, reset, formState } = useForm({
-    mode: "onTouched",
-  });
-  const { isSubmitted, isDirty } = formState;
-  const [selectedPrimaryProbeSetIds, setSelectedPrimaryProbeSetIds] = useState<
-    string[]
-  >(experiment?.primaryProbeSets?.map((p) => p?.id as string) || []);
-  const [
-    selectedSecondaryProbeSetsIds,
-    setselectedSecondaryProbeSetsIds,
-  ] = useState<string[]>(
-    experiment?.secondaryProbeSets?.map((p) => p?.id as string) || [],
+  const { probeSets } = useConfig();
+
+  const getProbeSetIds = (probeSets: ProbeSets) =>
+    probeSets?.map((probeSet) => probeSet?.id as string) || [];
+
+  // We must alter primary probe set options when a secondary set is selected
+  // to exclude the set from primary probe set options and vice versa
+  const [primaryProbeSetIds, setPrimaryProbeSetIds] = useState<string[]>(
+    getProbeSetIds(experiment?.primaryProbeSets!),
+  );
+  const [secondaryProbeSetIds, setSecondaryProbeSetIds] = useState<string[]>(
+    getProbeSetIds(experiment?.secondaryProbeSets!),
   );
 
-  const isValid = isServerValid && formState.isValid;
-  const isDirtyUnsaved = isDirty && !(isValid && isSubmitted);
+  const probeSetOption = (probeSet: ProbeSet) => ({
+    label: probeSet.name,
+    value: probeSet.id,
+  });
+
+  const primaryProbeSetOptions: SelectOption[] = [];
+  const secondaryProbeSetOptions: SelectOption[] = [];
+
+  // Get primary/secondary options from server-supplied array of probe sets
+  probeSets?.forEach((probeSet) => {
+    if (!secondaryProbeSetIds.includes(probeSet!.id)) {
+      primaryProbeSetOptions.push(probeSetOption(probeSet!));
+    }
+    if (!primaryProbeSetIds.includes(probeSet!.id)) {
+      secondaryProbeSetOptions.push(probeSetOption(probeSet!));
+    }
+  });
+
+  const defaultValues = {
+    primaryProbeSetIds:
+      experiment?.primaryProbeSets?.map((probeSet) =>
+        probeSetOption(probeSet!),
+      ) || "",
+    secondaryProbeSetIds:
+      experiment?.secondaryProbeSets?.map((probeSet) =>
+        probeSetOption(probeSet!),
+      ) || "",
+  };
+
+  const {
+    FormErrors,
+    formSelectAttrs,
+    isValid,
+    isDirtyUnsaved,
```

Comment:
It looks like across all of these forms `isDirtyUnsaved` is only needed for the `useExitWarning` check. Moving that hook inside your hook could further dry things up.

Reply:
Hmm, I think I'll file a follow up for this one, have a couple other things to look into. Great call out though.

Reply:
Follow up here: #4304